service/utils_test.go

```diff
@@ -124,3 +124,23 @@ func TestFormatSigningLogResponse(t *testing.T) {
 		t.Errorf("Expected the first fingerprint '%s', got: %s", signingLog[0].Fingerprint, result.SigningLog[0].Fingerprint)
 	}
 }
+
+func TestRandomGeneration(t *testing.T) {
+	// search for random strings enough smalls as to see if they are random
+	var tokens [10]string
+	for i := 0; i < 10; i++ {
+		// generate minimum amount of random data to verify it is enough random
+		token, err := GenerateRandomString(1)
+		if err != nil {
+			t.Errorf("Error generating random string: %v", err)
+		}
+		tokens[i] = token
+	}
+
+	// Check that the random strings look unique and valid
+	for i := 1; i < 10; i++ {
+		if tokens[i] == tokens[i-1] {
```

Comment:
This check looks as though it is only checking adjacent strings for duplication. So if the first and last string were the same, the test would still pass.

It would be better to make sure that none of the strings are duplicated. One approach is to create a map instead of an array:
`var tokens map[string]string`

Then you can check the number of entries instead:
`if len(tokens) < 10:`